src/librbd/ExclusiveLock.cc

```diff
@@ -134,11 +134,13 @@ void ExclusiveLock<I>::shut_down(Context *on_shut_down) {
   {
```

Comment:
i will try it, and i recorded the stack trace at:  
http://tracker.ceph.com/issues/19276